src/mango/src/mango_selector.erl

```diff
@@ -398,6 +400,14 @@ negate({[{Field, Cond}]}) ->
     {[{Field, negate(Cond)}]}.
 
 
+norm_deduplicate({[{Op, [H|T]}]}) ->
```

Comment:
I'm not seeing where this is called.

Reply:
It's a step in the selector normalization - (line 40 above). The intention is to deduplicate predicates (particularly `"$exists": true`) but this isn't necessary for correctness - it just makes the normalized selector cleaner. Thinking about it again, I'll remove it here and will consider improving selector normalization in a different PR.